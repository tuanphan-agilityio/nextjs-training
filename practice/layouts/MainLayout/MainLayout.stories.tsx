import { Story, Meta } from '@storybook/react';

import MainLayout, { MainLayoutProps } from '.';

export default {
  component: MainLayout,
  title: 'MainLayout',
} as Meta;

const Template: Story<MainLayoutProps> = (args) => <MainLayout {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: (
    <div>
      <h1>Content goes here</h1>
      <p>This is the main content of the layout.</p>
    </div>
  ),
};
