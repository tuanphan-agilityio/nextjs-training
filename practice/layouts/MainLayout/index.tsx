import { FC, ReactNode } from 'react';

import Header from '@/components/Header';
import Footer from '@/components/Footer';

interface MainLayoutProps {
  children: ReactNode;
}

const Layout: FC<MainLayoutProps> = ({ children }) => {
  return (
    <>
      <Header />
      {children}
      <Footer />
    </>
  );
};

export type { MainLayoutProps };

export default Layout;
