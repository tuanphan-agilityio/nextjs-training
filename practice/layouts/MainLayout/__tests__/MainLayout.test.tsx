import { render, screen } from '@testing-library/react';
import MainLayout, { MainLayoutProps } from '..';

describe('Main Layout', () => {
  const mockProps: MainLayoutProps = {
    children: 'Page content MainLayout',
  };

  it('displays the correct content', () => {
    render(<MainLayout {...mockProps} />);
    const pageContent = screen.getByText('Page content MainLayout');

    expect(pageContent).toBeInTheDocument();
  });

  it('renders correctly', () => {
    const { container } = render(<MainLayout {...mockProps} />);
    expect(container).toMatchSnapshot();
  });
});
