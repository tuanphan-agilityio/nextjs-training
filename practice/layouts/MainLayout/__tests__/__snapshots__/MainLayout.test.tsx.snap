// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Layout renders correctly 1`] = `
<div>
  <header
    class="h-20 shadow-lg text-secondary sm:h-14"
  >
    <div
      class="container flex justify-between items-center h-full"
    >
      <a
        aria-label="mangcoding Store"
        class="flex items-center text-sm font-quaternary-bold"
        href="/"
      >
        <svg
          class="mr-1"
          fill="none"
          height="34"
          width="36"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M35.046 0H.214A.215.215 0 0 0 0 .215V33.72c0 .12.096.216.214.216h34.832a.214.214 0 0 0 .214-.216V.215A.214.214 0 0 0 35.046 0Z"
            fill="#0EC5D7"
          />
          <path
            d="M9.32 7.468v18.999"
            stroke="#fff"
            stroke-width="3.082"
          />
          <path
            d="M15.012 7.468v9.463"
            stroke="#fff"
            stroke-width="3.076"
          />
          <path
            d="M19.349 7.469v3.214h5.432l3.94 8.143h3.55L26.634 7.469h-7.286Z"
            fill="#fff"
          />
          <path
            d="M20.963 14.167v12.677"
            stroke="#fff"
            stroke-width="3.56"
          />
        </svg>
         
        <span
          class="sm:hidden"
        >
          mangcoding Store
        </span>
      </a>
      <div
        class="flex gap-10 sm:gap-4"
      >
        <nav
          aria-label="header-navigation"
          class="lg:hidden"
        >
          <ul
            class="flex items-center gap-10 h-full font-secondary-regular text-md"
          >
            <li
              class="relative"
            >
              <a
                class="after:content-[""] after:absolute after:bottom-0 after:left-1/2 after:w-1 after:h-1 after:rounded-full after:bg-quaternary after:transform after:-translate-x-1/2 text-quaternary font-secondary-bold"
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="relative"
            >
              <a
                class="hover:text-quaternary"
                href="/shop"
              >
                Shop
              </a>
            </li>
            <li
              class="relative"
            >
              <a
                class="hover:text-quaternary"
                href="/about"
              >
                About Us
              </a>
            </li>
          </ul>
        </nav>
        <ul
          class="flex items-center gap-6 sm:flex-row-reverse sm:gap-1"
        >
          <li
            class="hover:cursor-pointer"
          >
            <svg
              fill="none"
              height="24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                stroke="#0D1222"
              >
                <path
                  d="M6.124 9.039a1.5 1.5 0 0 1 2.233-1.116l2.403 1.373a2.5 2.5 0 0 0 2.48 0l2.403-1.373a1.5 1.5 0 0 1 2.233 1.116l1.347 10.775a1.5 1.5 0 0 1-1.489 1.686H6.266a1.5 1.5 0 0 1-1.489-1.686L6.124 9.039Z"
                />
                <path
                  d="M15 6V4.01C15 2.896 14.008 2 12.799 2H11.2C9.981 2 9 2.906 9 4.01V6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-miterlimit="10"
                  stroke-width="1.2"
                />
                <circle
                  cx="9"
                  cy="12"
                  r=".8"
                  stroke-width=".4"
                />
                <circle
                  cx="15"
                  cy="12"
                  r=".8"
                  stroke-width=".4"
                />
              </g>
            </svg>
          </li>
          <li
            class="hover:cursor-pointer"
          >
            <svg
              class="sm:hidden"
              fill="none"
              height="24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.755 3c-2.092 0-3.901 1.28-4.75 3.115C11.156 4.28 9.357 3 7.255 3 4.355 3 2 5.444 2 8.454c0 4.31 6.043 9.837 8.732 12.082a1.965 1.965 0 0 0 2.536 0C15.967 18.28 22 12.754 22 8.454 22.01 5.444 19.655 3 16.755 3Z"
                stroke="#0D1222"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-miterlimit="10"
                stroke-width="1.2"
              />
            </svg>
          </li>
          <li
            class="hover:cursor-pointer"
          >
            <svg
              class="sm:hidden"
              fill="none"
              height="24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                stroke="#0D1222"
              >
                <circle
                  cx="12"
                  cy="7"
                  r="2.5"
                />
                <path
                  d="M5.5 19a6.5 6.5 0 1 1 13 0v.5h-13V19Z"
                />
              </g>
            </svg>
          </li>
          <li
            class="hover:cursor-pointer"
          >
            
            <div
              class="flex px-4 py-2 w-full bg-primary max-w-[250px] h-12 rounded-lg border border-1 border-solid border-tertiary sm:w-[160px] sm:h-10"
            >
              <input
                aria-label="search"
                class="block w-full border-none focus:outline-none text-left"
                placeholder="Search"
              />
              <div
                class="flex items-center w-6 bg-primary"
              >
                <svg
                  fill="none"
                  height="18"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="7.875"
                    cy="7.875"
                    r="5.063"
                    stroke="#969696"
                    stroke-width="1.125"
                  />
                  <rect
                    fill="#969696"
                    height="1"
                    rx=".5"
                    transform="rotate(45 11.869 11.162)"
                    width="5.25"
                    x="11.869"
                    y="11.162"
                  />
                </svg>
              </div>
            </div>
          </li>
        </ul>
        <div
          class="hidden lg:flex lg:items-center"
        >
          <svg
            height="1em"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 96c0-17.7 14.3-32 32-32h384c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zm0 160c0-17.7 14.3-32 32-32h384c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zm448 160c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32h384c17.7 0 32 14.3 32 32z"
            />
          </svg>
        </div>
      </div>
    </div>
  </header>
  Page content MainLayout
  <footer
    class="bg-footer-bg text-sm"
  >
    <div
      class="container flex gap-x-32 pt-20 pb-[92px] text-primary h-full xl:py-12 xl:gap-x-20 sm:flex-col sm:py-6"
    >
      <div
        class="w-1/3 sm:w-full sm:flex sm:flex-col sm:items-center"
      >
        <a
          aria-label="mangcoding Store"
          class="flex items-center text-sm font-quaternary-bold"
          href="/"
        >
          <svg
            class="mr-1"
            fill="none"
            height="34"
            width="36"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M35.046 0H.214A.215.215 0 0 0 0 .215V33.72c0 .12.096.216.214.216h34.832a.214.214 0 0 0 .214-.216V.215A.214.214 0 0 0 35.046 0Z"
              fill="#0EC5D7"
            />
            <path
              d="M9.32 7.468v18.999"
              stroke="#fff"
              stroke-width="3.082"
            />
            <path
              d="M15.012 7.468v9.463"
              stroke="#fff"
              stroke-width="3.076"
            />
            <path
              d="M19.349 7.469v3.214h5.432l3.94 8.143h3.55L26.634 7.469h-7.286Z"
              fill="#fff"
            />
            <path
              d="M20.963 14.167v12.677"
              stroke="#fff"
              stroke-width="3.56"
            />
          </svg>
           
          <span
            class="sm:hidden"
          >
            mangcoding Store
          </span>
        </a>
        <p
          class="mt-6 leading-6 font-primary-regular sm:text-center"
        >
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam mollis, justo nec porttitor auctor, erat sapien faucibus lectus, vel tempor dolor augue et lectus.
        </p>
        <div
          class="mt-10 bg-primary w-fit"
        >
          <button
            class="inline-flex justify-center items-center w-[180px] h-12 bg-primary font-primary-regular text-md hover:opacity-80 text-transparent font-primary-bold bg-clip-text bg-gradient-to-b from-primary-gradient-start to-primary-gradient-end text-primary bg-primary sm:w-[140px]"
          >
            Login Now
          </button>
        </div>
      </div>
      <div
        class="w-2/3 flex gap-x-20 lg:gap-x-16 lg:flex-wrap lg:gap-y-10 sm:mt-6 sm:w-full sm:gap-y-4 sm:justify-center"
      >
        <div>
          <p
            class="mb-6 text-md font-primary-bold sm:text-center"
          >
            Home
          </p>
          
          <ul
            class="flex flex-col gap-3"
          >
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Product
            </li>
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Categories
            </li>
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Shop
            </li>
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Login
            </li>
          </ul>
        </div>
        <div>
          <p
            class="mb-6 text-md font-primary-bold sm:text-center"
          >
            Shop
          </p>
          
          <ul
            class="flex flex-col gap-3"
          >
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              T-Shirt
            </li>
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Jacket
            </li>
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Shirt
            </li>
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Jens
            </li>
          </ul>
        </div>
        <div>
          <p
            class="mb-6 text-md font-primary-bold sm:text-center"
          >
            Categories
          </p>
          
          <ul
            class="flex flex-col gap-3"
          >
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Men
            </li>
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Children
            </li>
            <li
              class="hover:cursor-pointer sm:text-center text-sm"
            >
              Women
            </li>
          </ul>
        </div>
        <div>
          <p
            class="mb-6 text-md font-primary-bold sm:text-center"
          >
            Contact
          </p>
          <address
            class="mb-4 text-sm"
          >
            mangcoding@gmail.com
          </address>
          <ul
            class="flex gap-4"
          >
            <li
              class="hover:cursor-pointer sm:text-center flex justify-center items-center w-8 h-8 border border-primary"
            >
              <svg
                fill="none"
                height="14"
                width="7"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#a)"
                >
                  <path
                    d="M1.787 14V7.43H0V5.067h1.787v-2.02C1.787 1.457 2.842 0 5.275 0c.984 0 1.713.092 1.713.092L6.93 2.3s-.742-.007-1.553-.007c-.877 0-1.017.393-1.017 1.046v1.727H7L6.885 7.43H4.36V14H1.787Z"
                    fill="#fff"
                  />
                </g>
                <defs>
                  <clippath
                    id="a"
                  >
                    <path
                      d="M0 0h7v14H0z"
                      fill="#fff"
                    />
                  </clippath>
                </defs>
              </svg>
            </li>
            <li
              class="hover:cursor-pointer sm:text-center flex justify-center items-center w-8 h-8 border border-primary"
            >
              <svg
                height="1em"
                style="fill: #fff;"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                />
              </svg>
            </li>
            <li
              class="hover:cursor-pointer sm:text-center flex justify-center items-center w-8 h-8 border border-primary"
            >
              <svg
                height="1em"
                style="fill: #fff;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
                />
              </svg>
            </li>
            <li
              class="hover:cursor-pointer sm:text-center flex justify-center items-center w-8 h-8 border border-primary"
            >
              <svg
                height="1em"
                style="fill: #fff;"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
                />
              </svg>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
